app/classes/import/code_reader.rb

```diff
@@ -5,7 +5,8 @@ class CodeReader
   attr_reader :plot_code, :tree_code, :branch_code, :leaf_code, :suffix
 
   def initialize(code)
-    format = /^(\w+\d+)-((?:CSP\d+-\d+|\w\d+))(?:[-\s](\w+\d*\w*)(?:-(\w\d+\w?)(?:-(.+))?)?)?$/
+    format = /^(\w+\d+)-((?:CSP\d+-\d+|\w[\.\d]+))(?:[-\s](\w+\d*\w*)(?:-(\w\d+)(?:(\w\d+))?)?)?$/
```

Comment:
This is always a cluster fuck. I imagine one day this will just have to be broken out to regular logic.

...or the dreaded multiline regex.